### PR TITLE
Add sshd_settings job

### PR DESCRIPTION
**Pull request by dpb587**
For support configuring...

 * certificate authentication
 * disabling authorized_keys
 * disabling password authentication
 * disabling the printing of last log
 * host key overrides

Was using this in [ssh-conf](https://github.com/dpb587/ssh-conf-bosh-release), but was wondering if it'd be useful or appropriate in os-conf instead. Discuss.

### CODE REVIEW

File: jobs/sshd_settings/spec [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+name: sshd_settings
+templates:
+  bin/pre-start: bin/pre-start
+  etc/trusted_ca: etc/trusted_ca
+properties:
+  authorized_principals:
+    default: {}
+    description: "A hash of username (string) => principals ([]string) for restricting trusted CA-based access."
```
Comment (cppforlife) [inline jobs/sshd_settings/spec]: add `example`

File: jobs/sshd_settings/spec [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    description: "If true, disables user-specific authorized_keys files (requiring all SSH connections to be using signed public keys; this breaks `bosh ssh`!)."
+  disable_password_authentication:
+    default: false
+    description: "If true, disables password authentication (requiring all SSH connections to be public key-based)."
+  disable_print_last_log:
+    default: true
+    description: "If true, do not print information about recent logins."
+  host_key.rsa:
+    description: "Set the RSA host key of the server (this breaks `bosh ssh`)."
+  host_key.dsa:
```
Comment (cppforlife) [inline jobs/sshd_settings/spec]: let;s not allow dsa.

File: jobs/sshd_settings/templates/bin/pre-start [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+#
+# authorized_principals
+#
+
+dir=/var/vcap/jobs/sshd_settings/etc/authorized_principals
+rm -fr $dir
+
+<% if p('authorized_principals').length > 0 %>
+  mkdir $dir
```
Comment (cppforlife) [inline jobs/sshd_settings/templates/bin/pre-start]: explicit perms?

**Comment by drnic** [comment]
@dpb587 bump

**Comment by dpb587** [comment]
Forgot about this; I’ll try and revisit soon
-- 
Danny Berger
https://dpb587.me

**Comment by jhunt** [comment]
@dpb587 bumpity bump

**Comment by dpb587** [comment]
@jhunt sorry, I moved away from this feature and no longer need it to be able support it. I'm going to close, but feel free to pick it up in your own fork and make the requested changes and open a new one PR.

**Pull request closed without merging**